active/0004-integrating-extensible-resources-and-deployment-stacks.md

```diff
@@ -0,0 +1,547 @@
+
+## Terms and definitions
+
+### Deployemnt Stacks
+
+An Azure deployment stack is a type of Azure resource that enables the management of a group of Azure resources as an atomic unit. See the [documenation](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/deployment-stacks?tabs=azure-powershell) for details.
+
+### OAuth 2.0 On-behalf-of (OAuth OBO) flow
+
+The [Microsoft identity platform and OAuth 2.0 On-Behalf-Of flow](https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-on-behalf-of-flow), which describes the scenario of a web API using an identity other than its own to call another web API. Referred to as delegation in OAuth, the intent is to pass a user's identity and permissions through the request chain.
+
+The flow is a non-standard OAuth 2.0 extension developed by Microsoft.
+
+## Motivation
+
+Deployment Stacks offer users the convenience of managing a set of resources as a unified entity, enabling quicker updates and deletions. Currently, this functionality is limited to Azure resources due to the absence of a unique identifier (resource ID) for extensible resources and a standardized authentication mechanism across different extensibility providers.
+
+To overcome these limitations, the REP suggests modifications to the deployments API. A proposed change is the introduction of a `referenceID` field in `outputResources`, a freefrom string designed to uniquely identify extensible resources. Additionally, the proposal includes updates to the `deploymentProviders`, enabling it to store provider configurations. These configurations may include references to key vaults or resource properties, which can be utilized to access authentication details safely. These proposed enhancements allows Deployment Stacks to interact with extensibility providers to delete extensible resources.
+
+Implementing these changes ensure that extensible resources can be managed with the same level of efficiency and security as Azure resources within Deployment Stacks.
+
+## Detailed design
+
+### Reference ID
+
+> A reference ID is a URL-encoded, free-form string designed to uniquely identify an extensible resource. This ID is generated by extensibility providers and is returned as part of the responses from the `POST /resources/reference` or `POST /resources/preview` extensible resource APIs. The concept and design of reference IDs have been finalized.
+>
+> Any reference ID MUST adhere to the format `{flags}-{uniqueId}`.
+
+> - `{flags}`: This section consists of two binary digits indicating the type of reference ID. The currently accepted values are `00` and `01`:
+>   - `00`: signifies that the reference ID is autonomous and doesn't need extra data for GET and DELETE actions. This implies the utilization of the extensible resource APIs `GET /resources/{referenceId}` and `DELETE /resources/{referenceId}` for operations.
```

Comment:
I'm curious, and I might have missed a previous design session on the topic...

What's the difference between a provider that *defines an empty configuration* and a provider that *doesn't require configuration*?